### ISSUE TITLE
Incorrect entity search results interpretation while searching already existing items in Purview

### ISSUE THREAD

**Issue by uhrb** [microsoft/Purview-ADB-Lineage-Solution-Accelerator#50]
**Describe the bug**
Incorrect entity search results interpretation while searching already existing items in Purview

**To Reproduce**
1. Load any data from ADLSv2 (in my case it was https://datahub.io/core/airport-codes/) using augmented Databricks cluster, setup according to instructions and notebook on python/scala
2. Convert data, and compute several aggregates, write out aggregates into several delta lake tables, named same way, for example "airport-per-country", "airport-per-type", "airport-per-region"
3. Look at the purview - instead of 3 outputs of the notebook, it shows only one, named as first table wrote down to ADLSv2
4. 
**Expected behavior**
3 outputs, instead of 1

**Logs**
Code of function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs was modified, to add query logging to Purview
```
             var uid = Guid.NewGuid().ToString("N");
            _logger.LogInformation($"QUERY {uid} SEARCH {filter["filter"]}");
            List<QueryValeuModel> results = await this._client.Query_entities(filter["filter"]!);
            _logger.LogInformation($"QUERY {uid} RESULT {JsonConvert.SerializeObject(results, Formatting.None)}");

```
Output of this logging is 
```
2022-07-19T09:51:29.204 [Information] QUERY d7fd7d9195394014aac86cb64593b086 SEARCH {"and": [{"attributeName": "qualifiedName","operator": "contains","attributeValue": "https:%22%7D,%7B%22or%22: [{"attributeName": "qualifiedName","operator": "contains","attributeValue": "letarget.blob.core.windows.net"},{"attributeName": "qualifiedName","operator": "contains","attributeValue": "letarget.dfs.core.windows.net"}]},{"attributeName": "qualifiedName","operator": "contains","attributeValue": "containerout"},{"attributeName": "qualifiedName","operator": "contains","attributeValue": "delta-out"},{"attributeName": "qualifiedName","operator": "contains","attributeValue": "airport-per-type"}]}

2022-07-19T09:51:29.331 [Information] QUERY d7fd7d9195394014aac86cb64593b086 RESULT [{"owner":null,"qualifiedName":https://letarget.blob.core.windows.net/containerout/delta-out/airport-per-country,"entityType":"purview_custom_connector_generic_entity_with_columns","name":"airport-per-country","description":"Data Assets airport-per-country","term":null,"id":"cc4d36c7-865e-4f24-8817-fcf665e97138","label":null,"classification":null,"collectionId":null,"assetType":["Purview Custom Connector"],"@search.highlights":{"qualifiedName":["<em>https</em>://<em>letarget.blob.core.windows.net</em>/<em>containerout</em>/<em>delta</em>-<em>out</em>/<em>airport</em>-<em>per</em>-country"],"name":["<em>airport</em>-<em>per</em>-country"],"description":["Data Assets <em>airport</em>-<em>per</em>-country"]},"@search.score":3.2986124}]

```

As you can see, once Purview returns not "exact match", but suggestion, with different name, but code treats it like exact match

**Desktop (please complete the following information):**
 - OS: Azure
 - OpenLineage Version: 0.10
 - Databricks Runtime Version: 9.1LTS
 - Cluster Type:  Job
 - Cluster Mode: Single, Standard
 - Using Credential Passthrough: No

**Comment by wjohnson** [comment]
@uhrb Thank you for this detailed issue.

I'm able to reproduce this error and it seems that this will occur with hyphens `-` in the name but succeeds without hyphens.

@rodrigomonteiro-gbb  and I are taking a look into it and will look for a solution in the parser.

**Comment by wjohnson** [comment]
@uhrb - The code is updated on the release/2.0 and main branches.  Please checkout the release branch to test without including any additional experimental features we are including / hardening for the next minor version release.

You'll need to compile the function code and deploy it to your Azure function to get the latest version.

We plan on making a `2.0.2` release shortly that will enable deployment through the ARM template and point to the latest pre-compiled function code.

**Comment by wjohnson** [comment]
@uhrb please see the release 2.0.2 which includes the fix for this scenario.

**Comment by marktayl1** [comment]
Closing as resolved by 2.0.2
